test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/RazorDirectivesTest/DirectiveDescriptor_AllowsNullableTypes.stree.txt

```diff
@@ -1,76 +1,80 @@
-            SyntaxKind.Identifier;[KeyValuePair];
-            SyntaxKind.LessThan;[<];
-            SyntaxKind.Keyword;[string];
-            SyntaxKind.Comma;[,];
-            SyntaxKind.Whitespace;[ ];
-            SyntaxKind.Keyword;[string];
-            SyntaxKind.GreaterThan;[>];
-            SyntaxKind.QuestionMark;[?];
-            SyntaxKind.LeftBracket;[[];
-            SyntaxKind.RightBracket;[]];
-        Code span - Gen<None> - [ ] - SpanEditHandler;Accepts:Whitespace - (109:0,109) - Tokens:1
-            SyntaxKind.Whitespace;[ ];
-        Code span - Gen<DirectiveToken {;Type;Opt:False}> - [global::System.Collections.Generic.KeyValuePair<string, string>?[]] - DirectiveTokenEditHandler;Accepts:NonWhitespace - (110:0,110) - Tokens:18
-            SyntaxKind.Identifier;[global];
-            SyntaxKind.DoubleColon;[::];
-            SyntaxKind.Identifier;[System];
-            SyntaxKind.Dot;[.];
-            SyntaxKind.Identifier;[Collections];
-            SyntaxKind.Dot;[.];
-            SyntaxKind.Identifier;[Generic];
-            SyntaxKind.Dot;[.];
-            SyntaxKind.Identifier;[KeyValuePair];
-            SyntaxKind.LessThan;[<];
-            SyntaxKind.Keyword;[string];
-            SyntaxKind.Comma;[,];
-            SyntaxKind.Whitespace;[ ];
-            SyntaxKind.Keyword;[string];
-            SyntaxKind.GreaterThan;[>];
-            SyntaxKind.QuestionMark;[?];
-            SyntaxKind.LeftBracket;[[];
-            SyntaxKind.RightBracket;[]];
-    Markup span - Gen<Markup> - [] - SpanEditHandler;Accepts:Any - (176:0,176) - Tokens:1
-        SyntaxKind.Unknown;[];
+RazorDocument - [0..176)::176 - [@custom string? string?[] global::System.Int32? KeyValuePair<string, string>? KeyValuePair<string, string>?[] global::System.Collections.Generic.KeyValuePair<string, string>?[]]
+    MarkupBlock - [0..176)::176
+        MarkupTextLiteral - [0..0)::0 - [] - Gen<Markup> - SpanEditHandler;Accepts:Any
+            Marker;[];
+        CSharpCodeBlock - [0..176)::176
+            RazorDirective - [0..176)::176
+                CSharpTransition - [0..1)::1 - Gen<None> - SpanEditHandler;Accepts:None
+                    Transition;[@];
+                RazorDirectiveBody - [1..176)::175
+                    RazorMetaCode - [1..7)::6 - Gen<None> - SpanEditHandler;Accepts:None
+                        Identifier;[custom];
+                    CSharpCodeBlock - [7..176)::169
+                        CSharpStatementLiteral - [7..8)::1 - [ ] - Gen<None> - SpanEditHandler;Accepts:Whitespace
+                            Whitespace;[ ];
+                        CSharpStatementLiteral - [8..15)::7 - [string?] - Gen<DirectiveToken {;Type;Opt:False}> - DirectiveTokenEditHandler;Accepts:NonWhitespace
+                            Keyword;[string];
+                            QuestionMark;[?];
+                        CSharpStatementLiteral - [15..16)::1 - [ ] - Gen<None> - SpanEditHandler;Accepts:Whitespace
```

Comment:
Ya we'll also eventually want to change these whitespace markers to be a different type of significant

Reply:
Added to https://github.com/aspnet/Razor/issues/2619#issue-364659549